### PR TITLE
Implementation of Opcodes - Part 2

### PR DESCRIPTION
**Pull request by JereSalo**
# Summary
This is a second PR regarding to the implementation of the VM opcodes. This has all the necessary opcodes for the VM working except for the trap one.

### CODE REVIEW

File: src/instructions/add.rs [hunk 1]
```diff
@@ -30,7 +30,7 @@ impl VM {
 
 #[cfg(test)]
 mod tests {
-    use crate::vm::VM;
+    use super::*;
```
Comment (maximopalopoli) [inline src/instructions/add.rs]: Maybe you could have used `use super::VM` instead

File: src/instructions/br.rs [hunk 2]
```diff
@@ -1,5 +1,49 @@
...
-    pub fn op_br(&mut self, instr: u16) {}
+    pub fn op_br(&mut self, instr: u16) {
+        let nzp = (instr >> 9) & 0b111;
+        let pc_offset = instr & 0x1FF;
+
+        if nzp & self.reg.cond as u16 > 0 {
+            self.reg.pc = self.reg.pc.wrapping_add(sign_extend(pc_offset, 9));
+        }
+    }
+}
```
Comment (maximopalopoli) [inline src/instructions/br.rs:18]: I struggled to figure out what nzp means, maybe you can leave comments explaining, like you did in previous opcodes, or just improve the name

File: src/instructions/mod.rs [hunk 3]
```diff
@@ -22,10 +20,54 @@ fn sign_extend(mut x: u16, bit_count: u8) -> u16 {
...
+            8 => Ok(Opcode::OpRti),
+            9 => Ok(Opcode::OpNot),
+            10 => Ok(Opcode::OpLdi),
+            11 => Ok(Opcode::OpSti),
+            12 => Ok(Opcode::OpJmp),
+            13 => Ok(Opcode::OpRes),
+            14 => Ok(Opcode::OpLea),
+            15 => Ok(Opcode::OpTrap),
+            _ => Err(()),
+        }
```
Comment (maximopalopoli) [inline src/instructions/mod.rs:65]: Maybe you could leave in each opcode function a little explanation of what the operation does

**Review by maximopalopoli — COMMENTED** [review]
Hi, good job. I leave you some specific comments

File: src/instructions/add.rs (hunk 1 above)
Comment (JereSalo) [reply src/instructions/add.rs]: Done!

File: src/instructions/mod.rs (hunk 3 above)
Comment (JereSalo) [reply src/instructions/mod.rs:65]: Done! I left a very brief description of each.

File: src/instructions/br.rs (hunk 2 above)
Comment (JereSalo) [reply src/instructions/br.rs:18]: Done!

**Pull request merged**